pilot/pkg/config/kube/gateway/inferencepool_collection.go

```diff
@@ -356,3 +564,22 @@ func (c *Controller) canManageShadowServiceForInference(obj *corev1.Service) (bo
 	// We can manage if it has no manager or if we are the manager
 	return inferencePoolManaged, obj.GetResourceVersion()
 }
+
+func indexHTTPRouteByInferencePool(o *gateway.HTTPRoute) []string {
+	var keys []string
+	for _, rule := range o.Spec.Rules {
+		for _, backendRef := range rule.BackendRefs {
+			if (backendRef.BackendRef.Group != nil && string(*backendRef.BackendRef.Group) == gvk.InferencePool.Group) &&
+				(backendRef.BackendRef.Kind != nil && string(*backendRef.BackendRef.Kind) == gvk.InferencePool.Kind) {
+				// If BackendRef.Namespace is not specified, the backend is in the same namespace as the HTTPRoute's
+				backendRefNamespace := o.Namespace
+				if backendRef.BackendRef.Namespace != nil && *backendRef.BackendRef.Namespace != "" {
```

Comment:
Can we use `ptr.OrEmpty` to simplify these conditionals?

Reply:
Good idea

https://github.com/istio/istio/pull/56831/commits/9f2ae64f2cdd290b1bd2b025cbecd63f2d187540